MekHQ/src/mekhq/campaign/market/ContractMarket.java

```diff
@@ -160,33 +160,27 @@ public void generateContractOffers(Campaign campaign) {
     public void generateContractOffers(Campaign campaign, boolean newCampaign) {
         if (((method == ContractMarketMethod.ATB_MONTHLY) && (campaign.getLocalDate().getDayOfMonth() == 1))
                 || newCampaign) {
-			Contract[] list = contracts.toArray(new Contract[contracts.size()]);
-			for (Contract c : list) {
-				removeContract(c);
-			}
+            Contract[] list = contracts.toArray(new Contract[0]);
```

Comment:
Why are we sizing the array at 0?

Reply:
IDE suggestion, because if the size isn't consistent it'll just create the array anyway. However, I'll change it to be a forEach implementation, which will be a single line of code.